### PR TITLE
r2.1.0 ptl fix

### PR DESCRIPTION
**Pull request by pablo-garay**
r2.1.0 ptl fix

### CODE REVIEW

File: requirements/requirements_lightning.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 cloudpickle
 fiddle
 hydra-core>1.3,<=1.3.2
-lightning>2.2.1
+lightning>2.2.1,<=2.4.0
```
Comment (jstjohn) [inline requirements/requirements_lightning.txt:4]: Are you addressing the progress printer issue with 2.5? You can fix that directly by calling `on_train_epoch_start` from inside of a new `on_fit_start` in the progressprinter class.

Comment (jstjohn) [reply requirements/requirements_lightning.txt:4]: Eg you could add:

```python
    @override
    def on_fit_start(self, trainer: Trainer, pl_module: LightningModule) -> None:
        self.on_train_epoch_start(trainer, pl_module)
```

Right before https://github.com/NVIDIA/NeMo/blob/main/nemo/lightning/pytorch/callbacks/progress_printer.py#L114-L120

Comment (jstjohn) [reply requirements/requirements_lightning.txt:4]: that got 2.5 working for me with job resumption without the `AttributeError: 'ProgressPrinter' object has no attribute 'total'` error.

Comment (pablo-garay) [reply requirements/requirements_lightning.txt:4]: Virtually all tests are failing. Thats why . Needs workaround before 2.5 upgrade

**Comment by pablo-garay** [comment]
This change made tests pass again, so merging it

**Pull request merged**